### ISSUE TITLE
support for JSON-LD

### ISSUE THREAD

**Issue by jaygray0919** [Devs-Garden/jsonbase#5]
Can we use your database to manage an aggregation of JSON-LD statements?

**Comment by syamdanda** [comment]
Not supporting yet.

**Comment by melvincarvalho** [comment]
As JSON-LD is a subset of JSON, is it not already supported by the fact that JSON is supported?

**Comment by syamdanda** [comment]
We haven't tested this in those aspects yet, thats why I said not supported. will first work on adding the required enhancements and code reviews to release Alpha version. then ill look into all other aspects.
For sure we will check and let you know here..

**Comment by jaygray0919** [comment]
Can share test data when you are ready

**Comment by jaygray0919** [comment]
We have a number of 3rd normal form JSON-LD datasets. Google Structured Data Testing Tool properly joins the items to produce a fully de-normalized report. What we would like to do is specify (by query) the JSON-LD statements that are related to a target HTML page. Today, this is done manually. That is the 'test data' referred to above.

It would be very valuable if we could specify a page-specific query that injected the JSON-LD statements into that page.

Here's a wrinkle. Each of our statements is encoded in a `<script type='application/ld+json'></script>`
In other words, each statement is standalone.
When all statements have been properly concatenated, we have a processor that generates a single `<script>` where the individual statements are part of an `@graph`.
So the issue is: how to express (and store) a statement that can be queried?
None of the NoSQL databases give us that service (we know from direct experience).
That is the use case that we have in mind when we saw your announcement.

Let us know of you need input data to test a new version. We can provide two versions of the same collection of expressions: as multiple scripts and as a single @graph.

**Comment by melvincarvalho** [comment]
JSON-LD is just a subset of JSON

Are you putting each statement in a given document in its own `script` tag?

JSON-LD is designed to have many statements together in one structured data island.  You dont really need @graph you can just put them in an array and give each statement an @id

**Comment by syamdanda** [comment]
@jaygray0919  can you provide some test data and provide expected test results please.

**Comment by jaygray0919** [comment]
yes. gimme a day or two to pull together diverse configurations that will enable you to test corner cases.

**Comment by jaygray0919** [comment]
Here are 4 scenarios:

Scenario-1
Here is a report about the structure of a JSON-LD document: https://afdsi.org/sdl-2-NCIT-report/
Here is the source data used by the processor to generate the report: https://afdsi.org/sdl-2-NCIT-data/full-script-local-IRI.txt

---

Scenario-2
Here is report 2: https://afdsi.org/sdl-1-NALT-report/
Here is data for report 2: https://afdsi.org/sdl-1-NALT-data/data.txt

---

Scenario-3
Here is report-3: https://afdsi.org/sdl-3-GS1-report/
Here is data-3: https://afdsi.org/sdl-3-GS1-data/

---

Scenario-4
Here is data: https://afdsi.org/sdl-4-NALT-data/data.txt
Here is how it it is rendered on Google SDTT: https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fafdsi.org%2Fsdl-4-NALT-data%2Fdata.txt

---

Please ask me questions as you work on these examples; I can re-format them to be semantically equivalent but better fit your storage structure

/jay

**Comment by jaygray0919** [comment]
Are the examples useful, or do I need to put them into another format for your testing?

**Comment by syamdanda** [comment]
I think those are enough for initial implementation. will ask you my questions/data on-need basis
